### ISSUE TITLE
[REVIEW]: Agentpy: A package for agent-based modeling in Python

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3065]
**Submitting author:** @JoelForamitti (<a href="http://orcid.org/0000-0002-4828-7288">Joël Foramitti</a>)
**Repository:** <a href="https://github.com/JoelForamitti/agentpy" target ="_blank">https://github.com/JoelForamitti/agentpy</a>
**Version:** v0.1.1
**Editor:** @sbenthall
**Reviewer:** @jamesdamillington, @martibosch 
**Archive:** <a href="https://doi.org/10.5281/zenodo.4946508" target="_blank">10.5281/zenodo.4946508</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/6ed5785a714fe8c52c7b500d8fbdc8a5/status.svg)](https://joss.theoj.org/papers/6ed5785a714fe8c52c7b500d8fbdc8a5)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/6ed5785a714fe8c52c7b500d8fbdc8a5"><img src="https://joss.theoj.org/papers/6ed5785a714fe8c52c7b500d8fbdc8a5/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/6ed5785a714fe8c52c7b500d8fbdc8a5/status.svg)](https://joss.theoj.org/papers/6ed5785a714fe8c52c7b500d8fbdc8a5)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/JoelForamitti/agentpy" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@jamesdamillington & @martibosch , please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @sbenthall know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/JoelForamitti/agentpy">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@JoelForamitti) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/JoelForamitti/agentpy">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@JoelForamitti) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @jamesdamillington, @martibosch  it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
Failed to discover a `Statement of need` section in paper

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.41 s (144.7 files/s, 165391.1 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Python                          27            776            923           2113
reStructuredText                18            208            144            455
Jupyter Notebook                 6              0          62148            386
Markdown                         2             24              0             84
TeX                              1              8              1             70
YAML                             2              1              1             34
DOS Batch                        1              8              1             26
make                             1              4              7              9
CSS                              1              1              2              6
-------------------------------------------------------------------------------
SUM:                            59           1030          63227           3183
-------------------------------------------------------------------------------


Statistical information for the repository '4f497de77da3cb9525eb4407' was
gathered on 2021/02/26.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
JoelForamitti                   13           208            128            2.44
Joël Foramitti                  44          8561           4859           97.56

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
JoelForamitti              3812         1832.7          1.8                9.76
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1177/0963721419834547 is OK
- 10.3389/fevo.2018.00237 is OK

MISSING DOIs

- 10.1038/460685a may be a valid DOI for title: The economy needs agent-based modelling
- 10.1057/jos.2009.28 may be a valid DOI for title: Managing business complexity: discovering strategic solutions with agent-based modeling and simulation

INVALID DOIs

- https://doi.org/10.1002/wcc.647 is INVALID because of 'https://doi.org/' prefix
- https://doi.org/10.1002/wics.1356 is INVALID because of 'https://doi.org/' prefix
- https://doi.org/10.1016/j.cosrev.2017.03.001 is INVALID because of 'https://doi.org/' prefix
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03065/joss.03065/10.21105.joss.03065.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03065/joss.03065/10.21105.joss.03065.pdf) :page_facing_up: :point_left:

**Comment by sbenthall** [comment]
@JoelForamitti One thing you can always do while waiting for reviews is correct the DOI's noted as missing or invalid by the **whedon** bot. Those will need to be resolved before publication.

**Comment by JoelForamitti** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1177/0963721419834547 is OK
- 10.3389/fevo.2018.00237 is OK
- 10.1038/460685a is OK
- 10.1002/wcc.647 is OK
- 10.1002/wics.1356 is OK
- 10.1016/j.cosrev.2017.03.001 is OK
- 10.1093/acprof:oso/9780195172119.001.0001 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @jamesdamillington, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @martibosch , please update us on how your review is going (this is an automated reminder).

**Comment by jamesdamillington** [comment]
Apologies, yes this is still on my todo list. Teaching a priority, but this will follow very soon

**Comment by martibosch** [comment]
Hello! I have my thesis defense the 30th of March, so I will get to this review after that. Sorry for the inconveniences, I hope that this timing still works for you.

**Comment by sbenthall** [comment]
Hello. Gentle reminder to @martibosch and @jamesdamillington to please review this work.

**Comment by JoelForamitti** [comment]
@jamesdamillington, thank you for your comment! From other publications in JOSS (see e.g. [this recent one](https://joss.theoj.org/papers/10.21105/joss.03097)) I had the impression that it is allowed to explain the need without an explicit section title of that name. @sbenthall, could you confirm or refute this?

**Comment by jamesdamillington** [comment]
@JoelForamitti @sbenthall I've completed my review at [JOSS Review (JM)](https://github.com/JoelForamitti/agentpy/issues/13) and conditionally accept the submission depending on the issues highlighted there. Looks good! Any queries on my comments, just let me know.

**Comment by JoelForamitti** [comment]
Thanks a lot @jamesdamillington. I will wait for the second review from @martibosch as well as a response from @sbenthall on the 'statement of need' section before starting to work on a revision!

**Comment by sbenthall** [comment]
I'm actually not sure myself. I'll need to ask somebody with more experience. @xuanxu can you answer this question about Statement of Need?

**Comment by xuanxu** [comment]
Yes, the `Statement of need` is a required section and we just [recently](https://github.com/openjournals/joss/pull/883) [changed](https://github.com/openjournals/joss/pull/862) the documentation to better refelct this and avoid exceptions like the mentioned paper.

**Comment by JoelForamitti** [comment]
Ok, thanks! I will include it in the revised version

**Comment by sbenthall** [comment]
Hello @martibosch this is a gentle reminder that this submission is waiting for your review

**Comment by martibosch** [comment]
Hello all! I am sorry for my delay, these were (and still are) some crazy days with the end of my thesis and some proposal deadlines to find further funding opportunities.

I have finished my review, I find the package to be very neat and easy (enjoyable) to use, which is great. The examples are clear and properly cover all the features except the scenario comparison, which is why I submitted [a dedicated issue](https://github.com/JoelForamitti/agentpy/issues/15). Besides this, I only have some minor corrections in the manuscript which are addressed in [another issue](https://github.com/JoelForamitti/agentpy/issues/16).

Overall, I find that agentpy is a great piece of work, my remarks are minor and should be quite easy to address. Looking forward to use it for my research!

Best,
Martí

**Comment by JoelForamitti** [comment]
Thank you a lot @martibosch, I will start working on a revision now!

**Comment by sbenthall** [comment]
@JoelForamitti Hello. How is progress going?

**Comment by JoelForamitti** [comment]
@sbenthall thanks for keeping us on track! I am aiming to finish my revision by the end of this week

**Comment by JoelForamitti** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03065/joss.03065/10.21105.joss.03065.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03065/joss.03065/10.21105.joss.03065.pdf) :page_facing_up: :point_left:

**Comment by JoelForamitti** [comment]
Dear @sbenthall, @jamesdamillington, & @martibosch,

above you can find the revised manuscript for my submission. it comes together with a new version of the software (v0.1.0) that has been released today. i have tried to take all the comments from the review into account, and also added some new parts to include the new features that have been released since the original submission in January.

**Comment by martibosch** [comment]
I believe that all the issues that I raised have been addressed hence from my side, the manuscript can move forward to publication.

**Comment by sbenthall** [comment]
Super.

@martibosch could you please check off all the boxes in your review checklist?

@jamesdamillington what do you think of the revisions? Do they satisfy you?

**Comment by martibosch** [comment]
I just have one last question regarding the "Summary" checkbox. I believe that the second paragraph of the "Introduction" of the current manuscript provides a concise summary of agentpy, however, do JOSS papers require a dedicated "Summary" section? If not, I can check off the box and the manuscript can proceed to publication.

**Comment by jamesdamillington** [comment]
Hi All, I was away last week. I will check the revisions this week.

**Comment by jamesdamillington** [comment]
The revisions look good to me (and I have closed [the issue](https://github.com/JoelForamitti/agentpy/issues/13) I opened). I did [wonder the same things](https://github.com/openjournals/joss-reviews/issues/3065#issuecomment-854723095) as @martibosch about the 'summary', but I also agree that the second paragraph of the Introduction section is a good summary if a dedicated Summary section is not needed (hence I have checked the Summary checklist item). I am happy that this can move to publication.

**Comment by JoelForamitti** [comment]
Fantastic, thank you! @sbenthall I have made a final commit to add acknowledgments to the paper

**Comment by JoelForamitti** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03065/joss.03065/10.21105.joss.03065.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03065/joss.03065/10.21105.joss.03065.pdf) :page_facing_up: :point_left:

**Comment by JoelForamitti** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03065/joss.03065/10.21105.joss.03065.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03065/joss.03065/10.21105.joss.03065.pdf) :page_facing_up: :point_left:

**Comment by JoelForamitti** [comment]
@sbenthall I have added the citation

**Comment by xuanxu** [comment]
> Is it necessary to have a dedicated "Summary" section, or is having the summary in the introduction sufficient?

@sbenthall The paper should include a summary, but is not mandatory to title the section as 'Summary', it is recommended, but 'Introduction' is also OK.

**Comment by sbenthall** [comment]
Super. This paper passes review.

We can move on with the next steps!

**Comment by sbenthall** [comment]
@whedon generate pdf

**Comment by sbenthall** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1177/0963721419834547 is OK
- 10.3389/fevo.2018.00237 is OK
- 10.1038/460685a is OK
- 10.1002/wcc.647 is OK
- 10.1002/wics.1356 is OK
- 10.1016/j.cosrev.2017.03.001 is OK
- 10.1093/acprof:oso/9780195172119.001.0001 is OK
- 10.1038/s42254-020-00273-3 is OK
-  10.25080/Majora-7b98e3ed-009  is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03065/joss.03065/10.21105.joss.03065.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03065/joss.03065/10.21105.joss.03065.pdf) :page_facing_up: :point_left:

**Comment by sbenthall** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1177/0963721419834547 is OK
- 10.3389/fevo.2018.00237 is OK
- 10.1038/460685a is OK
- 10.1002/wcc.647 is OK
- 10.1002/wics.1356 is OK
- 10.1016/j.cosrev.2017.03.001 is OK
- 10.1093/acprof:oso/9780195172119.001.0001 is OK
- 10.1038/s42254-020-00273-3 is OK
-  10.25080/Majora-7b98e3ed-009  is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by sbenthall** [comment]
I've proofread the article. It looks great. I'm also excited for the publication of this work -- this is software I could use in my work!

@JoelForamitti could you please make a new tagged release and archive it using Zenoda, and then report the version number and archive DOI in this thread?

**Comment by JoelForamitti** [comment]
@sbenthall thank you for your kind words!
here is the new release:

Version: v0.1.1
DOI: 10.5281/zenodo.4946508
Link: https://doi.org/10.5281/zenodo.4946508

**Comment by JoelForamitti** [comment]
@sbenthall is there anything else that needs to be done from my side?

**Comment by sbenthall** [comment]
@whedon set 10.5281/zenodo.4946508 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4946508" target="_blank">10.5281/zenodo.4946508</a> is the archive.

**Comment by sbenthall** [comment]
@whedon set v0.1.1as version

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by sbenthall** [comment]
@whedon set v0.1.1 as version

**Comment by whedon** [comment]
OK. v0.1.1 is the version.

**Comment by sbenthall** [comment]
I recommend this package for acceptance!

**Comment by sbenthall** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1177/0963721419834547 is OK
- 10.3389/fevo.2018.00237 is OK
- 10.1038/460685a is OK
- 10.1002/wcc.647 is OK
- 10.1002/wics.1356 is OK
- 10.1016/j.cosrev.2017.03.001 is OK
- 10.1093/acprof:oso/9780195172119.001.0001 is OK
- 10.1038/s42254-020-00273-3 is OK
-  10.25080/Majora-7b98e3ed-009  is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2405

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2405, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kyleniemeyer** [comment]
Everything looks good to me, proceeding to publish!

**Comment by kyleniemeyer** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1407471800964063233) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2406
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03065](https://doi.org/10.21105/joss.03065)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by kyleniemeyer** [comment]
Congratulations @JoelForamitti on your article's publication in JOSS!

Many thanks to @jamesdamillington and @martibosch for reviewing this, and @sbenthall for editing.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03065/status.svg)](https://doi.org/10.21105/joss.03065)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03065">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03065/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03065/status.svg
   :target: https://doi.org/10.21105/joss.03065
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03065/status.svg)](https://doi.org/10.21105/joss.03065)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by JoelForamitti** [comment]
Fantastic, thanks again to everyone!

**Comment by JoelForamitti** [comment]
Hi @kyleniemeyer, I realized there is typo in the acknowledgements! Is there a way to still fix this?

**Comment by kyleniemeyer** [comment]
@JoelForamitti yes, I think if you update the paper source, @arfon can update the published version

**Comment by arfon** [comment]
@JoelForamitti – that should be updated now. The updated paper might take a few hours to show up properly on the JOSS site due to caching.

**Comment by JoelForamitti** [comment]
@arfon fantastic, thanks a lot!
